### PR TITLE
Updates TwoPartyTradeFlow to use the CollectSignaturesFlow. Updates t…

### PR DESCRIPTION
**Pull request by joeldudleyr3**
…ests to not generate fresh keys (will re-implement post identity work).

### CODE REVIEW

File: finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt [hunk 1]
```diff
@@ -70,57 +61,43 @@ object TwoPartyTradeFlow {
...
             }
+            val stx = subFlow(signTransactionFlow)
+
+            // Wait for the finished, notarised transaction to arrive in our transaction store.
+            progressTracker.currentStep = WAITING_FOR_TX
+            return waitForLedgerCommit(stx.id)
         }
+        // DOCEND 4
+
+        // DOCSTART 5
```
Comment (roger3cev) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: What happens here? There was no 5th example?

Comment (roger3cev) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: Ahh. DOCSTART 5 is repeated here.

File: finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt [hunk 2]
```diff
@@ -70,62 +61,42 @@ object TwoPartyTradeFlow {
...
+            return subFlow(signTransactionFlow)
+            // DOCEND 5
         }
-        // DOCEND 5
+        // DOCEND 4
 
         // Following comment moved here so that it doesn't appear in the docsite:
-        // There are all sorts of funny games a malicious secondary might play with it sends maybeSTX (in
-        // receiveAndCheckProposedTransaction), we should fix them:
+        // There are all sorts of funny games a malicious secondary might play with it sends maybeSTX,
```
Comment (roger3cev) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: Where's maybeSTX now?

File: finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt [hunk 3]
```diff
@@ -165,16 +136,16 @@ object TwoPartyTradeFlow {
             // Put together a proposed transaction that performs the trade, and sign it.
             progressTracker.currentStep = SIGNING
             val (ptx, cashSigningPubKeys) = assembleSharedTX(tradeRequest)
-            val stx = signWithOurKeys(cashSigningPubKeys, ptx)
+            val partSignedTx = signWithOurKeys(cashSigningPubKeys, ptx)
```
Comment (roger3cev) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: The method could be removed right? It's only one line!

File: docs/source/flow-state-machines.rst [hunk 4]
```diff
@@ -243,87 +241,111 @@ Let's implement the ``Seller.call`` method. This will be run when the flow is in
...
+- The party we wish to send the message to.
+- The payload being sent.
+
+``send`` will serialise the payload and send it to the other party automatically.
+
+Next, we call a *subflow* called ``SignTransactionFlow`` (see :ref:`subflows`). ``SignTransactionFlow`` automates the
+process of:
+
+* Receiving a proposed trade transaction from the buyer, with the buyer's signature attached.
+* Checking that the proposed transaction is valid.
```
Comment (roger3cev) [inline docs/source/flow-state-machines.rst]: We can add that you have to override the checkProposal method to ensure that it IS the transaction you expect to receive!

File: docs/source/flow-state-machines.rst [hunk 5]
```diff
@@ -345,122 +367,124 @@ Flows can be composed via nesting. Invoking a sub-flow looks similar to an ordin
...
 
 * Send the transaction to the chosen notary and, if necessary, satisfy the notary that the transaction is valid.
 * Record the transaction in the local vault, if it is relevant (i.e. involves the owner of the node).
 * Send the fully signed transaction to the other participants for recording as well.
 
+.. warning:: If the seller stops before sending the finalised transaction to the buyer, the seller is left with a
+   valid transaction but the buyer isn't, so they can't spend the asset they just purchased! This sort of thing is not
+   always a risk (as the seller may not gain anything from that sort of behaviour except a lawsuit), but if it is, a future
+   version of the platform will allow you to ask the notary to send you the transaction as well, in case your counterparty
+   does not. This is not the default because it reveals more private info to the notary.
```
Comment (roger3cev) [inline docs/source/flow-state-machines.rst]: we can drop the "more" :)

Comment (joeldudleyr3) [inline docs/source/flow-state-machines.rst]: addressed

File: finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt (hunk 3 above)
Comment (joeldudleyr3) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: addressed - no, as `assembleSharedTX` returns a pair that must be destructured before being passed into `signWithOurKeys`

File: finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt (hunk 2 above)
Comment (joeldudleyr3) [inline finance/src/main/kotlin/net/corda/flows/TwoPartyTradeFlow.kt]: addressed - thanks, good catch

File: docs/source/flow-state-machines.rst (hunk 4 above)
Comment (joeldudleyr3) [inline docs/source/flow-state-machines.rst]: addressed

**Pull request merged**